packages/web/src/builders/file-server/file-server.impl.ts

```diff
@@ -125,38 +102,78 @@ export default async function* fileServerExecutor(
   }
   run();
 
-  const outputPath = getBuildTargetOutputPath(opts, context);
-  const args = getHttpServerArgs(opts);
+  const outputPath = getBuildTargetOutputPath(options, context);
 
-  const serve = exec(`npx http-server ${outputPath} ${args.join(' ')}`, {
-    cwd: context.root,
-  });
-  const processExitListener = () => serve.kill();
-  process.on('exit', processExitListener);
-  serve.stdout.on('data', (chunk) => {
-    if (chunk.toString().indexOf('GET') === -1) {
-      process.stdout.write(chunk);
+  const app = connect();
+
+  app.use(serveStatic(outputPath, { fallthrough: true }));
+
+  if (options.proxyUrl) {
+    app.use(createProxyMiddleware({ target: options.proxyUrl }));
+  } else {
+    // SPA Fallback to index.html
+    app.use((_req, res, next) => {
```

Comment:
🤔 I think we shouldn't just do the index.html fallback when there is no proxy. With the webpack dev server for example, you can easily have a proxy configuration, but still have the fallback part.

We might want to look at the configured proxy URLs. Maybe we can reuse the webpack devserver logic or middleware that is being used there for providing the same behavior.

Reply:
@juristr I will take a look at this this week. I will go through the sourcecode of the webpack-dev-middleware package maybe we can use it directly or just parts of it.